lib/encode.js

```diff
@@ -13,39 +21,61 @@ import {inspect} from './util.js';
  *   CBOR-LD bytes.
  * @param {documentLoaderFunction} options.documentLoader -The document loader
  *   to use when resolving JSON-LD Context URLs.
- * @param {boolean} [options.compressionMode=1] - `1` to use compression mode
- *   version 1, `0` to use no compression.
- * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
- *   their encoded CBOR-LD values (must be values greater than 32767 (0x7FFF)).
- * @param {diagnosticFunction} [options.diagnose] - A function that, if
- * provided, is called with diagnostic information.
- *
+ * @param {number} options.varintValue - The varint value for the
+ *  registry entry associated with the resulting CBOR-LD payload. For legacy
+ *  support, use varintTagValue = -1.
+ * @param {Map} options.keywordsTable -A map of JSON-LD keywords
+ *  and their associated CBOR-LD integer values.
+ * @param {Map} options.stringTable - A map of string values and
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.urlSchemeTable - A map of URL schemes to
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.typedLiteralTable - A map of JSON-LD types
+ *  to maps that map values of that type to their associated CBOR-LD
+ *  integer values.
```

Comment:
I will convert this into an issue for the future:

We should consider some simpler API that will do the registry lookup based on the `varintValue` (to be renamed potentially to something like `registryEntryId` ... I dunno) and get the right tables that way. A function could be provided that can fetch those tables -- and the only advantage of using that approach here would be to help avoid mismatches; and maybe that's not worth the extra effort.